SonarFunctions.cmake

```
@@ -2,7 +2,7 @@ include(GetGitRevisionDescription)
 
 function(sonar_cpack_version major_var minor_var patch_var)
   git_describe(version --match "v*")
-  set(regex "v([0-9]+)\.([0-9]+)\.([0-9]+)(.*)")
+  set(regex "^v([0-9]+)\\.([0-9]+)\\.([0-9]+)(-.*|)$")
```

Can this work? I am trying to remove the dash from the pre-release group
```suggestion
  set(regex "^v([0-9]+)\\.([0-9]+)\\.([0-9]+)(-(.*))?$")
```

---

The reason I didn't want to do that was I wasn't sure how to conditionally add a `-` into the final version based on if that capture group matches or not. If you know how that works, then that should be fine.

---

I see. Okay, go with your version then. Let's wrap this up.

---

Does that mean merge this?

---

yeah :)